include/elliptics/utils.hpp

```diff
@@ -342,10 +342,15 @@ class argument_data_base
 	}
 
 	argument_data_base(const char *data) :
-		m_data(pointer_type::from_raw(const_cast<char *>(data), std::strlen(data)))
+		m_data(pointer_type::from_raw(const_cast<char *>(data), std::strlen(data))) // It isn't correct for binary data.
 	{
 	}
 
+        argument_data_base(const char *data, size_t size) :
```

Comment:
Maybe you should use here `const void *data` to explicitly specify that the constructor is for binary data.